model/change_point_detection.go

```diff
@@ -387,16 +398,46 @@ func GetTotalPagesForChangePointsGroupedByVersion(ctx context.Context, env cedar
 		if err != nil {
 			return 0, errors.Wrap(err, "Cannot decode response of getting count of change points grouped by version")
 		}
-		return int(math.Ceil(float64(res.Count) / float64(pageSize))), nil
+		return int(math.Ceil(float64(res.Count) / float64(args.PageSize))), nil
 	}
-	return 0, errors.New("Not able to get count of total changepoints matching query")
+	return 0, nil
 }
 
-func appendAfterBaseGetChangePointsByVersionAgg(projectId string, additionalSteps ...bson.M) []bson.M {
+func appendAfterBaseGetChangePointsByVersionAgg(args GetChangePointsGroupedByVersionOpts, additionalSteps ...bson.M) []bson.M {
+	matchStage := bson.M{
+		bsonutil.GetDottedKeyName(perfInfoKey, perfResultInfoProjectKey):  args.ProjectId,
+		bsonutil.GetDottedKeyName(perfInfoKey, perfResultInfoVariantKey):  bson.M{"$regex": args.VariantRegex},
+		bsonutil.GetDottedKeyName(perfInfoKey, perfResultInfoVersionKey):  bson.M{"$regex": args.VersionRegex},
+		bsonutil.GetDottedKeyName(perfInfoKey, perfResultInfoTaskNameKey): bson.M{"$regex": args.TaskRegex},
+		bsonutil.GetDottedKeyName(perfInfoKey, perfResultInfoTestNameKey): bson.M{"$regex": args.TestRegex},
+	}
```

Comment:
do we have anything to assert that these regexes are valid or anchored? are these queries well indexed?

Reply:
We have no requirements regarding the regex as of yet. One hopes that the ~10 users of this API who work for us won't be too naughty. Are there certain cases that worry you specifically that we'd want to handle here right now? All that comes to mind based on my brief reading of $regex in mdb is that you wouldn't want case insensitive regexes or such, but we're not allowing user to pass options anyway.

W.r.t. indices, last I checked we have indices for project id, variant, version, task, and test.

Reply:
Hope springs eternal! Also thanks for checking on the indexes. 

I believe if you do not anchor the regexes (e.g. have them start with ^) then the queries can't use the indexes,) so if we don't validate for that in the application then we could end up impacting the DB with a query, which seems not-ideal.

Reply:
Just did a quick check here: https://docs.mongodb.com/manual/reference/operator/query/regex/

Seems like as long as it's not case insensitive, it will leverage an index; Left anchoring/prefixes allow for an additional optimization of only traversing the range of that index with such a prefix.

Reply:
I think it's well enough that it uses the index without requiring a prefix match -- e.g. its probably a common use case to search for "latency" and want to find things like "average_latency_90th_percentile".